### ISSUE TITLE
Ocl for AdaptiveLIFRate

### ISSUE THREAD

**Issue by MenelikV** [nengo/nengo_ocl#85]
Hello everybody,

As I was told in the previous [thread](https://github.com/nengo/nengo_ocl/issues/84), `nengo_ocl` does not support `AdaptiveLIFRate`.
I looked the code on the github and I came up with this modifications to support this neuron class (based on the code for `LIFRate` in `nengo_ocl` and `AdaptiveLIFRate`).

Could someone correct me if I am wrong (I hope not ^^) ?

(nb: the modifications did not return any errors as I tested my own model)

Here are all the modifications I made :

* In `sim_ocl.py`
  
```python
from nengo_ocl.clra_nonlinearities plan_adaptive_lif_rate
from nengo.neurons import AdaptiveLIFRate```
```
  
```python
def _plan_AdaptiveLIFRate(self, ops): 
    J = self.all_data[[self.sidx[op.J] for op in ops]]
    R = self.all_data[[self.sidx[op.output] for op in ops]]
    ref = self.RaggedArray(
        [np.array(op.neurons.tau_ref, dtype=J.dtype) for op in ops])
    tau = self.RaggedArray(
        [np.array(op.neurons.tau_rc, dtype=J.dtype) for op in ops])
    dt = self.model.dt
    tau_adapt = self.RaggedArray(
        [np.array(op.neurons.tau_n, dtype=J.dtype) for op in ops])
    inc = self.RaggedArray(
        [np.array(op.neurons.inc_n, dtype=J.dtype) for op in ops])

   	return [plan_adaptive_lif_rate(self.queue, J, R, ref, tau, dt, tau_adapt, inc,
                              n_elements=2)]
```

```python
def plan_SimNeurons(self,all_ops): 
    groups = groupby(all_ops, lambda op: op.neurons.__class__)
    plans = []
    for neuron_class, ops in groups:
    	if neuron_class is LIF:
    		plans.extend(self._plan_LIF(ops))
    	elif neuron_class is LIFRate:
    		plans.extend(self._plan_LIFRate(ops))
    	elif neuron_class is AdaptiveLIFRate:
    		plans.extend(self._plan_AdaptiveLIFRate(ops))
    	else:
    		raise ValueError("Unsupported neuron type '%s'"
                                 % neuron_class.__name__)
    return plans
```
* in `clra_nonlinearities.py`
 ```python
def plan_adaptive_lif_rate(queue, J, R, ref, tau, dt, tau_adapt, inc, tag=None, n_elements=0):
    assert R.ctype == J.ctype

    inputs = dict(j=J)
    outputs = dict(r=R)
    parameters = dict(tau=tau, ref=ref,tau_adapt = tau_adapt, inc = inc)
    textconf = dict(Rtype=R.ctype)
    declares = """
        const ${Rtype} c0 = 0, c1 = 1;
        """
    text = """
        j = max(j - inc, c0);
        r = c1 / (ref + tau * log1p(c1/j));
        inc += (dt/tau_adapt)*(inc*(r-n))
        """
    declares = as_ascii(
        Template(declares, output_encoding='ascii').render(**textconf))
    return _plan_template(
        queue, "cl_adaptive_lif_rate", text, declares=declares,
        tag=tag, n_elements=n_elements,
        inputs=inputs, outputs=outputs, parameters=parameters)
```

Thanks for your time and your tips :D

Menelik.

**Comment by hunse** [comment]
Looks good! Can you make a pull request? Don't forget to add your name to `CONTRIBUTORS.rst`.

**Comment by hunse** [comment]
Implemented in #88.
